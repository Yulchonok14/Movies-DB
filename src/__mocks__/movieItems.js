export const movieItems = [
    {
        id: 111,
        title: 'Bloodshot',
        date: '2020',
        rating: '3.6',
        genre: 'Action, Drama, Sci-Fi',
        imageUrl: 'https://upload.wikimedia.org/wikipedia/ru/9/9d/Bloodshot_poster.jpg',
        description: 'Vin Diesel fans who can’t wait for the next installment of the “Fast and Furious” macho soap opera series can get their fix at “Bloodshot,” a comic book adaptation that’s as big a stickler about “family” yet far less satisfying than even the worst films of the "Fast" franchise. The family in question here is the wife of Ray Garrison (Diesel), who is put in danger by her spouse’s mercenary soldiering. Now, if you want to walk into director Dave Wilson’s sci-fi actioner as blindly as I did, exit this review now. If you desire a hint of what you’re in for, let me leave you with a few phrases you would have encountered had you stuck around: “Universal Soldier,” “robotic cucarachas,” “needle drop abuse of the Talking Heads” and “blatant rip-off.”',
        duration: '110'
    },
    {
        id: 222,
        title: 'The invisible man',
        date: '2019',
        rating: '4.6',
        genre: 'Horror, Sci-Fi, Thriller',
        imageUrl: 'https://avatars.mds.yandex.net/get-kinopoisk-image/1900788/a90dd0dd-a8ae-4112-9865-23eb8f5879a8/360',
        description: 'Vin Diesel fans who can’t wait for the next installment of the “Fast and Furious” macho soap opera series can get their fix at “Bloodshot,” a comic book adaptation that’s as big a stickler about “family” yet far less satisfying than even the worst films of the "Fast" franchise. The family in question here is the wife of Ray Garrison (Diesel), who is put in danger by her spouse’s mercenary soldiering. Now, if you want to walk into director Dave Wilson’s sci-fi actioner as blindly as I did, exit this review now. If you desire a hint of what you’re in for, let me leave you with a few phrases you would have encountered had you stuck around: “Universal Soldier,” “robotic cucarachas,” “needle drop abuse of the Talking Heads” and “blatant rip-off.”',
        duration: '165'
    },
    {
        id: 333,
        title: 'Monster trucks',
        date: '2019',
        rating: '3.9',
        genre: 'Adventure, Action, Comedy',
        imageUrl: 'https://upload.wikimedia.org/wikipedia/ru/5/5e/Monster_Trucks_%28film%29.jpg',
        description: 'Vin Diesel fans who can’t wait for the next installment of the “Fast and Furious” macho soap opera series can get their fix at “Bloodshot,” a comic book adaptation that’s as big a stickler about “family” yet far less satisfying than even the worst films of the "Fast" franchise. The family in question here is the wife of Ray Garrison (Diesel), who is put in danger by her spouse’s mercenary soldiering. Now, if you want to walk into director Dave Wilson’s sci-fi actioner as blindly as I did, exit this review now. If you desire a hint of what you’re in for, let me leave you with a few phrases you would have encountered had you stuck around: “Universal Soldier,” “robotic cucarachas,” “needle drop abuse of the Talking Heads” and “blatant rip-off.”',
        duration: '154'
    },
    {
        id: 444,
        title: 'Cold comes the night',
        date: '2016',
        rating: '4.4',
        genre: 'Drama, Thriller, Crime',
        imageUrl: 'https://laughingsquid.com/wp-content/uploads/2014/01/CCTN_poster_Billy_HD-for-web.jpg',
        description: 'Vin Diesel fans who can’t wait for the next installment of the “Fast and Furious” macho soap opera series can get their fix at “Bloodshot,” a comic book adaptation that’s as big a stickler about “family” yet far less satisfying than even the worst films of the "Fast" franchise. The family in question here is the wife of Ray Garrison (Diesel), who is put in danger by her spouse’s mercenary soldiering. Now, if you want to walk into director Dave Wilson’s sci-fi actioner as blindly as I did, exit this review now. If you desire a hint of what you’re in for, let me leave you with a few phrases you would have encountered had you stuck around: “Universal Soldier,” “robotic cucarachas,” “needle drop abuse of the Talking Heads” and “blatant rip-off.”',
        duration: '129'
    },
    {
        id: 555,
        title: 'Snowden',
        date: '2010',
        rating: '4.8',
        genre: 'Drama, Thriller, Biography',
        imageUrl: 'https://upload.wikimedia.org/wikipedia/ru/1/1a/Snowden_%28film%29.jpg',
        description: 'Vin Diesel fans who can’t wait for the next installment of the “Fast and Furious” macho soap opera series can get their fix at “Bloodshot,” a comic book adaptation that’s as big a stickler about “family” yet far less satisfying than even the worst films of the "Fast" franchise. The family in question here is the wife of Ray Garrison (Diesel), who is put in danger by her spouse’s mercenary soldiering. Now, if you want to walk into director Dave Wilson’s sci-fi actioner as blindly as I did, exit this review now. If you desire a hint of what you’re in for, let me leave you with a few phrases you would have encountered had you stuck around: “Universal Soldier,” “robotic cucarachas,” “needle drop abuse of the Talking Heads” and “blatant rip-off.”',
        duration: '113'
    },
    {
        id: 666,
        title: 'Knock knock',
        date: '2011',
        rating: '2.6',
        genre: 'Horror, Thriller',
        imageUrl: 'https://i1.wp.com/morbidlybeautiful.com/wp-content/uploads/2015/10/kk-2.jpg',
        description: 'Vin Diesel fans who can’t wait for the next installment of the “Fast and Furious” macho soap opera series can get their fix at “Bloodshot,” a comic book adaptation that’s as big a stickler about “family” yet far less satisfying than even the worst films of the "Fast" franchise. The family in question here is the wife of Ray Garrison (Diesel), who is put in danger by her spouse’s mercenary soldiering. Now, if you want to walk into director Dave Wilson’s sci-fi actioner as blindly as I did, exit this review now. If you desire a hint of what you’re in for, let me leave you with a few phrases you would have encountered had you stuck around: “Universal Soldier,” “robotic cucarachas,” “needle drop abuse of the Talking Heads” and “blatant rip-off.”',
        duration: '187'
    }
];