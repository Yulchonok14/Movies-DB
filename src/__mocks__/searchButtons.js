export const searchByButtonsArr = [
    {id: 111, text: 'title', isActive: true},
    {id: 222, text: 'genre', isActive: false}
];