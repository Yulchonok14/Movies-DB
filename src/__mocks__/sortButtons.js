export const sortByButtonsArr = [
    {id: 111, text: 'release date', isActive: true},
    {id: 222, text: 'rating', isActive: false}
];