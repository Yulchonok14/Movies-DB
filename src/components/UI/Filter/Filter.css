.Filter {
    display: flex;
    color: #FFFFFF;
    text-transform: uppercase;
    margin: 15px 0;
    align-items: center;
}

.Filter div {
    border-radius: 10px;
    margin: 0 20px;
}

.Filter button {
    padding: 10px 20px;
    text-transform: uppercase;
    color: #FFFFFF;
    background-color: #424242;
    border: none;
    opacity: 0.8;
    outline: none;
}

.Filter button:hover {
    box-shadow: 9px 1px 11px rgba(192,192,192,192.85), 8px 10px 10px rgba(192,192,192,192.82);
}

.Filter .Active {
    background-color: #F65261;
    opacity: 1;
}

.Filter button:first-child {
    border-radius: 5px 0 0 5px;
}

.Filter button:last-child {
    border-radius: 0 5px 5px 0;
}

@media (min-width: 600px) {
    .Filter {
        margin: 0;
    }
}