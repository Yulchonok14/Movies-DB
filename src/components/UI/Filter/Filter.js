import React from 'react';
import PropTypes from 'prop-types';

import classes from './Filter.css';

const Filter = (props) => {
    const filterBtns = props.buttons
        ?
        props.buttons.map((btn, index) =>
            <button
                className={btn.isActive ? classes.Active: null}
                key={btn.id}
                onClick={()=>props.clicked(index)}>
                {btn.text}
            </button>)
        : null;
    return (
        <div style={{display: props.show ? 'flex' : 'none'}}
             className={classes.Filter}>
            <label>{props.label}</label>
            <div>{filterBtns}</div>
        </div>
    )
};

Filter.propTypes = {
    clicked: PropTypes.func,
    label: PropTypes.string,
    show: PropTypes.bool,
    buttons: PropTypes.arrayOf(PropTypes.shape({
        id:PropTypes.number.isRequired,
        text:PropTypes.string.isRequired,
        isActive: PropTypes.bool.isRequired
    }))
};

export default Filter;