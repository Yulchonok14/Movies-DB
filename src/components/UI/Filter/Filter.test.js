import React from 'react';

import { configure, shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';

import Filter from './Filter';
import { sortByButtonsArr } from '../../../__mocks__/sortButtons';

configure({
    adapter: new Adapter()
});

describe('<Filter/>', () => {
    const mockCallBack = jest.fn();
    let wrapper;

    beforeEach(() => {
        wrapper = shallow(<Filter label="Label"/>);
    });

    it('Filter should have buttons', () => {
        wrapper.setProps({
            clicked: mockCallBack,
            show: true,
            label: 'Label',
            buttons: sortByButtonsArr
        });
        expect(wrapper.find('button')).toHaveLength(2);
    });

    it('Filter should not have buttons', () => {
        wrapper.setProps({
            clicked: mockCallBack,
            show: false,
            label: 'Label',
            buttons: []
        });
        expect(wrapper.contains(<label>Label</label>)).toBeTruthy();
    });

    it('Upon the button click callback should be called', () => {
        wrapper.setProps({
            clicked: mockCallBack,
            show: true,
            label: 'Label',
            buttons: sortByButtonsArr
        });
        wrapper.find('button').first().simulate('click');
        expect(mockCallBack).toHaveBeenCalled();
    });

    it('Filter renders correctly', () => {
        const tree = renderer
            .create(<Filter clicked={mockCallBack}
                            show
                            label={'Label'}
                            buttons={sortByButtonsArr}/>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});