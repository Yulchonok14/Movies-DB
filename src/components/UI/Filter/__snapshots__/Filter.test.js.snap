// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Filter/> Filter renders correctly 1`] = `
<div
  style={
    Object {
      "display": "flex",
    }
  }
>
  <label>
    Label
  </label>
  <div>
    <button
      onClick={[Function]}
    >
      release date
    </button>
    <button
      className={null}
      onClick={[Function]}
    >
      rating
    </button>
  </div>
</div>
`;
